packages/core-dialog/core-dialog.js

```diff
@@ -1 +1,128 @@
+import {name, version} from './package.json'
+import {IS_BROWSER, queryAll, addEvent, dispatchEvent} from '../utils'
 
+const UUID = `data-${name}-${version}`.replace(/\W+/g, '-') // Strip invalid attribute characters
+const SUPPORT = IS_BROWSER && typeof window.HTMLDialogElement !== 'undefined'
+const BACKDROP = IS_BROWSER && document.createElement('backdrop')
+const PREVIOUS = `${UUID}-previous`
+const KEYS = {ESC: 27, TAB: 9}
+
+const FOCUSABLE_ELEMENTS = `
+  [tabindex]:not([disabled]),
+  a:not([disabled]),
+  button:not([disabled]),
+  input:not([disabled]),
+  select:not([disabled]),
+  textarea:not([disabled])`
+
+export default function dialog (dialogs, open) {
+  const options = typeof open === 'object' ? open : {open}
+
+  return queryAll(dialogs).forEach((dialog) => {
+    dialog.setAttribute(UUID, '')
+    dialog.setAttribute('role', 'dialog')
+    dialog.setAttribute('aria-modal', true)
+    dialog.setAttribute('aria-label', dialog.getAttribute('aria-label') || options.label)
+
+    toggleDialog(dialog, options.open)
+    return dialog
+  })
+}
+
+addEvent(UUID, 'click', (event) => {
+  for (let el = event.target, isClose; el; el = el.parentElement) {
+    const action = el.getAttribute('data-core-dialog')
+    isClose = isClose || action === 'close'
```

Comment:
Maybe comment why we do this loop - "close action needs to know the context of which dialog it is inside" or something :)